plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/PyStringCodeCompletion.java

```diff
@@ -149,17 +160,61 @@
     /**
      * Needed interface for adding the completions on a request
      * @throws MisconfigurationException
+     * @throws PythonNatureWithoutProjectException
+     * @throws IOException
      */
     @Override
     public TokensOrProposalsList getCodeCompletionProposals(CompletionRequest request) throws CoreException,
-            BadLocationException, MisconfigurationException {
+            BadLocationException, MisconfigurationException, IOException, PythonNatureWithoutProjectException {
+
         List<ICompletionProposalHandle> completionProposals = new ArrayList<>();
         request.showTemplates = false; //don't show templates in strings
         fillWithEpydocFields(request, completionProposals);
 
         TokensOrProposalsList ret = new TokensOrProposalsList();
         if (completionProposals.size() == 0) {
             //if the size is not 0, it means that this is a place for the '@' stuff, and not for the 'default' context for a string.
+            IDocument doc = request.doc;
+            ITypedRegion partition = ((FastPartitioner) PyPartitionScanner.checkPartitionScanner(doc))
+                    .getPartition(request.documentOffset);
+            String partitionType = partition.getType();
+
+            if (IPythonPartitions.F_STRING_PARTITIONS.contains(partitionType)) {
+                // Now we are going to check whether where we are in the given completion offset
+                int requestOffset = request.documentOffset;
+                int partitionOffset = partition.getOffset();
+                int partitionLine = doc.getLineOfOffset(partitionOffset);
+
+                int extraCol = 0;
+                // If the line that FString starts is the same line on which completion was requested
+                // we will need to set an extra column from where FString starts
+                if (partitionLine == request.getLine()) {
+                    extraCol = partitionOffset - doc.getLineOffset(partitionLine);
```

Comment:
You need to add the extraCol based on the `node` from `getBalancedExpressionsToBeEvaluatedInRegularGrammar` (otherwise you are adding the `extraCol` to all the lines and this may make you try to get a bad location in the document).